src/components/CurrencySelectionList/index.tsx

```diff
@@ -39,17 +66,34 @@ function CurrencySelectionList({searchInputLabel, initiallySelectedCurrencyCode,
             });
         }
 
-        return {
-            sections: isEmpty
-                ? []
-                : [
-                      {
-                          data: filteredCurrencies,
-                      },
-                  ],
-            headerMessage: isEmpty ? translate('common.noResultsFound') : '',
-        };
-    }, [currencyList, searchValue, canSelectMultiple, translate, initiallySelectedCurrencyCode, selectedCurrencies]);
+        if (shouldDisplaySelectedOptionOnTop) {
+            result.push({
+                title: '',
+                data: selectedOptions,
+                shouldShow: true,
+            });
+        }
+
+        if (shouldDisplayRecentlyOptions) {
+            const cutPolicyRecentlyUsedCurrencyOptions = policyRecentlyUsedCurrencyOptions.slice(0, CONST.IOU.MAX_RECENT_REPORTS_TO_SHOW);
+            if (!isEmpty) {
+                result.push(
+                    {
+                        title: translate('common.recents'),
+                        data: shouldDisplaySelectedOptionOnTop ? getUnselectedOptions(cutPolicyRecentlyUsedCurrencyOptions) : cutPolicyRecentlyUsedCurrencyOptions,
+                        shouldShow: shouldDisplayRecentlyOptions,
+                    },
+                    {title: translate('common.all'), data: shouldDisplayRecentlyOptions ? getUnselectedOptions(filteredCurrencies) : filteredCurrencies},
```

Comment:
NAB: Could we assign a variable getUnselectedOptions(filteredCurrencies) above and use it, as I see the same call used twice?